internal/pkg/global/config.go

```diff
@@ -476,6 +479,71 @@ func ensureDefaults() {
 	}
 }
 
+// userLookup is interface to enable mocking os/user package.
+type userLookup interface {
+	Current() (*user.User, error)
+	LookupGroupID(gid string) (*user.Group, error)
+}
+
+type osUserLookup struct {
+	*user.User
+}
+
+func (o *osUserLookup) Current() (*user.User, error) {
+	return user.Current()
+}
+
+func (o *osUserLookup) LookupGroupID(gid string) (*user.Group, error) {
+	return user.LookupGroupId(gid)
+}
+
+var getUserGroupIds = func(u *user.User) ([]string, error) {
+	return u.GroupIds()
+}
+
+// systemdCanBeActivated returns true if agent user is part of systemd-journal group. Returns false
+// and the error if one occurs. Used to deactivate systemd node discovery path.
+func systemdCanBeActivated(usr userLookup, targetGrp string) (bool, error) {
+	u, err := usr.Current()
+	if err != nil {
+		return false, err
+	}
+
+	gids, err := getUserGroupIds(u)
+	if err != nil {
+		return false, err
+	}
+
+	for _, gid := range gids {
+		grp, err := usr.LookupGroupID(gid)
+		if err != nil {
+			continue
```

Comment:
Just double checking if this is the intended flow: if usr.LookupGroupID() returns error, this function returns false without logging or exposing an error.

Reply:
I don't think so. In case of an error (i.e. gid not found) the error is ignored and the next gid is checked.